diff -urN zip-2.3.orig/fileio.c zip-2.3/fileio.c
--- zip-2.3.orig/fileio.c	Sun Nov  7 11:29:03 1999
+++ zip-2.3/fileio.c	Sat May  6 00:56:35 2000
@@ -919,66 +919,3 @@
 
 #endif /* NO_RENAME */
 
-
-#ifdef ZMEM
-
-/************************/
-/*  Function memset()   */
-/************************/
-
-/*
- * memset - for systems without it
- *  bill davidsen - March 1990
- */
-
-char *
-memset(buf, init, len)
-register char *buf;     /* buffer loc */
-register int init;      /* initializer */
-register unsigned int len;   /* length of the buffer */
-{
-    char *start;
-
-    start = buf;
-    while (len--) *(buf++) = init;
-    return(start);
-}
-
-
-/************************/
-/*  Function memcpy()   */
-/************************/
-
-char *
-memcpy(dst,src,len)             /* v2.0f */
-register char *dst, *src;
-register unsigned int len;
-{
-    char *start;
-
-    start = dst;
-    while (len--)
-        *dst++ = *src++;
-    return(start);
-}
-
-
-/************************/
-/*  Function memcmp()   */
-/************************/
-
-int
-memcmp(b1,b2,len)                     /* jpd@usl.edu -- 11/16/90 */
-register char *b1, *b2;
-register unsigned int len;
-{
-
-    if (len) do {       /* examine each byte (if any) */
-      if (*b1++ != *b2++)
-        return (*((uch *)b1-1) - *((uch *)b2-1));  /* exit when miscompare */
-    } while (--len);
-
-    return(0);          /* no miscompares, yield 0 result */
-}
-
-#endif  /* ZMEM */
diff -urN zip-2.3.orig/zip.h zip-2.3/zip.h
--- zip-2.3.orig/zip.h	Mon Nov  8 20:36:51 1999
+++ zip-2.3/zip.h	Sat May  6 00:55:56 2000
@@ -60,6 +60,7 @@
 
 /* Set up portability */
 #include "tailor.h"
+#include <strings.h>
 
 #ifdef USE_ZLIB
 #  include "zlib.h"
@@ -433,12 +434,6 @@
 int setfileattr OF((char *, int));
 char *tempname OF((char *));
 int fcopy OF((FILE *, FILE *, ulg));
-
-#ifdef ZMEM
-   char *memset OF((char *, int, unsigned int));
-   char *memcpy OF((char *, char *, unsigned int));
-   int memcmp OF((char *, char *, unsigned int));
-#endif /* ZMEM */
 
         /* in system dependent fileio code (<system>.c) */
 #ifndef UTIL
